### DISCUSSION TITLE
Optimize initial lense model for `DESIJ1205+4110`

### DISCUSSION THREAD

**Discussion by mirsazzathossain** [AstroBridge/BDLensing#31] [General]
I wanted to share a notebook that I created for fitting a lens model to the galaxy cluster `DESIJ1205+4110`. You can find the notebook [here](https://github.com/mirsazzathossain/BDLensing/blob/e03d7f1d700b31e2c22f762b50a67227220ba7a3/DESIJ1205%2B4110/lens_model_DESIJ1205%2B4110.ipynb).

At the moment, I have obtained a reduced chi-square value of approximately **1.569**, indicating that the model fits the data somewhat well. However, I'm still looking for ways to improve the fit even further. I would appreciate any suggestions or feedback on how to refine the model.

Please feel free to comment with any ideas, suggestions, or feedback. I look forward to hearing from you! 

Best,
@mirsazzathossain

**Comment by ajshajib** [comment]
The `gamma` value became atypically low. Let's fix it to 2 for now while tweaking other things. The source light is trying to contribute to the light at the center of the galaxy, which means the galaxy's light model is not adequate with one Sersic function. So, I suggest adding a second Sersic function for the lensing galaxy's light model, fixing the `n_sersic`s to 1 and 4, and joining the centroids of these 2 Sersics. Double Sersic is commonly needed to adequately model lens galaxy's light profile, for example, this paper: https://ui.adsabs.harvard.edu/abs/2021MNRAS.503.2380S/abstract.

**Comment by ajshajib** [comment]
Also, there are some strong blue residuals right outside the borders of some of the small circular masked regions, which means those masks need to be a little larger to cover up those pixels with blue residuals.

**Comment by mirsazzathossain** [comment]
@ajshajib bhiya, I have added a second Sersic function to the lens galaxy's light model and increased the radius of the smaller circular mask. However, I noticed that even when I make the mask bigger, there are still pixels with a blue radius outside the mask. I'm not sure why this is happening and would appreciate any insights or suggestions from you.

Additionally, I fixed the value of gamma in the model, but it seems to be having a hard time reconstructing the upper arc. I'm still exploring ways to improve the fit, but in the meantime, I would appreciate any feedback or suggestions on how to better handle these issues.

You can check out the updated notebook with the changes [here](https://github.com/mirsazzathossain/BDLensing/blob/3326057844fcb709c171635d854d73621060ce9d/DESIJ1205%2B4110/lens_model_DESIJ1205%2B4110.ipynb).

**Reply by ajshajib** [reply]
This system is quite complex, and it may be possible that a simple model (like the one we're trying now) wouldn't work until we individually add mass components for those smaller galaxies nearby. However, you may try disjoining the centroid and ellipticity of the mass and light and see what you get.

**Reply by mirsazzathossain** [reply]
Hi @ajshajib bhiya, after disjoining the centroid and ellipticity of the mass and light, I was able to improve the fit and achieve a reduced $\chi^2$ value of 1.19. You can check out the updated notebook with the changes [here](https://github.com/mirsazzathossain/BDLensing/blob/952c269b239ac8d358a413e6f706b0cf79ce8e8d/DESIJ1205%2B4110/lens_model_DESIJ1205%2B4110.ipynb). I'd love to hear your thoughts and insights on this model.

**Reply by ajshajib** [reply]
Looks great! I find it particularly interesting that now the model is predicting one of the masked blobs as a lensed image (at the end of the top arc). 👀 What happens when you unmask that blob now? 🤔

What's the distance now between the mass and light centroids in the arcsecond unit and in the kpc unit, assuming a redshift of 0.5?

**Reply by mirsazzathossain** [reply]
I tried unmasking the blob at the edge of the upper arc, but it resulted in a lot of residual left in both the upper and lower arcs. Therefore, I think it would be best to keep this blob masked. However, I will try making it slightly smaller to see what effect it has on the model.

You can check out the updated notebook [here](https://github.com/mirsazzathossain/BDLensing/blob/4f5c8b988967e2099e0fea1ac1d995c470ef7650/DESIJ1205%2B4110/lens_model_DESIJ1205%2B4110.ipynb). Also, I added the results of the distance calculation between the mass and light centroids of the previous model that you had asked for at the end of this notebook.

![output](https://user-images.githubusercontent.com/26181151/232623366-0e7baadd-070a-4bf6-b7bc-f1202c8e4fef.png)
![output2](https://user-images.githubusercontent.com/26181151/232623371-2ee18e9a-6c77-4857-886d-c61da5e74f29.png)

**Reply by ajshajib** [reply]
Ok, interesting. What does it look like if you plot the model from the previous nice model fit, but in the `ModelPlot()`, you provide a mask with that blob unmasked instead of the one used in fitting?

**Reply by mirsazzathossain** [reply]
Here is how it initially looked:
![before_res](https://user-images.githubusercontent.com/26181151/233328445-277da1d0-7621-4610-a388-e89367273178.png)
![before](https://user-images.githubusercontent.com/26181151/233326003-7d7ecbd6-5fac-45c8-951a-6004fd45dc4f.png)

After removing the mask of the blob at the edge of the upper arc:
![after_res](https://user-images.githubusercontent.com/26181151/233326265-dd7c559f-17ed-45bd-8de0-dd6f5a257297.png)
![after_model](https://user-images.githubusercontent.com/26181151/233326281-71bd4155-4212-4e79-937f-31d018e27491.png)

**Comment by mirsazzathossain** [comment]
@ajshajib bhiya, I've attempted to include the mass and light profile of the neighboring galaxy and my reduced $\chi^2$ value is now $1.45$ and the structure of the arcs are also not satisfactory. Could you please take a look at [my notebook](https://github.com/mirsazzathossain/BDLensing/blob/c3daa9fb96cdd252dca06ce4fbe59c99542a2ded/lens_systems/DESIJ1205%2B4110/lens_model_DESIJ1205%2B4110.ipynb) and verify if I correctly added the new profiles? Additionally, if you could suggest any improvements based on the updated fit, I would appreciate it.

**Reply by ajshajib** [reply]
The central galaxies mass profile ended up having a very large `center_y`, so I suggest making the bound on the centroid parameters much smaller than ±10". Don't add the mass profile for the neighboring galaxy first, only add its light profile, and then see what the model residual looks like. If you add the mass profile for the neighboring galaxy later, set the upper limit of its `theta_E` to a smaller value like 0.5".
